Source/SwiftLintCore/Models/Linter.swift

```diff
@@ -62,8 +62,13 @@ private extension Rule {
               configuration: Configuration,
               superfluousDisableCommandRule: SuperfluousDisableCommandRule?,
               compilerArguments: [String]) -> LintResult? {
+        // We shouldn't lint if the current Swift version is not supported by the rule
+        guard SwiftVersion.current >= Self.description.minSwiftVersion else {
+            return nil
+        }
+
         // Empty files shouldn't trigger violations
-        guard !file.isEmpty, SwiftVersion.current >= Self.description.minSwiftVersion else {
+        guard !shouldLintEmptyFiles, !file.isEmpty else {
```

Comment:
Does `shouldLintEmptyFiles` mean "lint **only** empty files"? If so, the name should make that clear and this condition should use XOR, shouldn't it? So I think then it should be:
```suggestion
        if file.isEmpty XOR !shouldLintEmptyFiles {
```

Reply:
> Does shouldLintEmptyFiles mean "lint only empty files"?

I don't think so. For example, in my case I'd like to also lint against files that only have comments (so they're not "empty" per the definition in `file.isEmpty`)